tests/BuildScriptGenerator.Tests/DotnetCore/GlobalJsonSdkResolverTest.cs

```diff
@@ -0,0 +1,1217 @@
+            IEnumerable<string> inputVersions,
+            IEnumerable<string> expectedSortedVersions)
+        {
+            // Arrange
+            var originalList = inputVersions.Select(sdk => SdkVersionInfo.Parse(sdk));
+
+            // Act
+            var actual = originalList.OrderBy(sdk => sdk);
+
+            // Assert
+            var actualSortedVersions = actual.Select(sdk => sdk.RawString);
+            Assert.Equal(expectedSortedVersions, actualSortedVersions, StringComparer.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void Disable_ReturnsExactVersion_IfFound()
+        {
+            // Arrange
+            var expectedVersion = "3.1.100";
+            var availableVersions = new[] { "2.1.100", "3.1.100", "3.1.101", "3.1.102-preview1-03444" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Disable,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Theory]
+        [InlineData("true")]
+        [InlineData(null)]
+        public void Disable_ReturnsExactPrereleaseVersion_IfFound(string allowPrerelease)
+        {
+            // Arrange
+            var expectedVersion = "3.1.102-preview1-03444";
+            var availableVersions = new[] { "3.1.100", "3.1.101", "3.1.102-preview1-03444", "3.1.102-preview2-03444" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.102-preview1-03444",
+                RollForward = GlobalJsonSdkResolver.Disable,
+                AllowPreRelease = allowPrerelease,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Equal(expectedVersion, actual);
+        }
+
+        [Fact]
+        public void Disable_ReturnsNull_IfExactVersionIsNotFound()
+        {
+            // Arrange
+            var availableVersions = new[] { "2.1.100", "3.1.101" };
+            var globalJson = new GlobalJsonModel();
+            globalJson.Sdk = new Sdk
+            {
+                Version = "3.1.100",
+                RollForward = GlobalJsonSdkResolver.Disable,
+            };
+            var globalJsonHelper = GetGlobalJsonHelper();
+
+            // Act
+            var actual = globalJsonHelper.GetSatisfyingSdkVersion(globalJson, availableVersions);
+
+            // Assert
+            Assert.Null(actual);
+        }
+
+        [Fact]
+        public void Patch_ReturnsSameVersion_IfPresent_AndDoesNotGetLatestPatchVersion()
```

Comment:
Maybe we could come up with a naming rules for test class names, like `SomeFeature_**Returns**What_**If**WhatHappens`. In my opinion, this one could be `Patch_ReturnsSameVersion_IfPresentsAndDoesNotReturnLatestPatchVersion`.

Reply:
Sure, sounds good.